stock_quant_package_dimension/models/stock_quant_package.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class StockQuantPackage(models.Model):
+    _inherit = "stock.quant.package"
+
+    pack_weight = fields.Float("Weight (kg)")
+    # lngth IS NOT A TYPO: https://github.com/odoo/odoo/issues/41353
+    lngth = fields.Integer("Length (mm)", help="length in millimeters")
+    width = fields.Integer("Width (mm)", help="width in millimeters")
+    height = fields.Integer("Height (mm)", help="height in millimeters")
+    volume = fields.Float(
+        "Volume (m³)",
+        digits=(8, 4),
+        compute="_compute_volume",
+        readonly=True,
+        store=False,
+        help="volume in cubic meters",
+    )
+
+    @api.depends("lngth", "width", "height")
+    def _compute_volume(self):
+        for pack in self:
+            pack.volume = (pack.lngth * pack.width * pack.height) / 1000.0 ** 3
+
+    def auto_assign_packaging(self):
+        self = self.with_context(_auto_assign_packaging=True)
```

Comment:
I am not a fan of passing arguments to methods via the context. In versions <13.0, this type of approach has a significant impact on performance since it invalidates the cache. However, the way the code is extended does not leave us any other solution.  For me the current approach is not good in the sense that the context of an environment relates to the data and should be dissociated from the context necessary for method execution. I wonder if it would be interesting to develop a decorator that would allow us to specify elements useful for an execution context...
ex:
```python
from odoo.addons.call_context import call_context

    ...
    
    def auto_assign_packaging(self):
        with call_context(_auto_assign_packaging=True):
            return super(StockQuantPackage, self).auto_assign_packaging()

    def write(self, vals):
        res = super(StockQuantPackage, self).write(vals)
        if call_context.get("_auto_assign_packaging") and vals.get(
            "product_packaging_id"
        ):
            self._update_dimensions_from_packaging(override=False)
       return res

```